care/facility/api/viewsets/patient_external_test.py

```diff
@@ -143,6 +143,13 @@ def bulk_upsert(self, request, *args, **kwargs):
             raise ValidationError({"sample_tests": "No Data was provided"})
         if not isinstance(request.data["sample_tests"], list):
             raise ValidationError({"sample_tests": "Data should be provided as a list"})
+
+        # check if the user is from same district
+        user = User.objects.filter(username=request.user).first()
+        for data in request.data["sample_tests"]:
+            if user.district != data["district"]:
```

Comment:
```suggestion
        for data in request.data["sample_tests"]:
            if request.user.district != data["district"]:
```
any specific reason to query the user again?

Reply:
oh, I forgot request.user is an object